src/schemathesis/models.py

```diff
@@ -259,13 +262,135 @@ class Check:
     message: Optional[str] = attr.ib(default=None)  # pragma: no mutate
 
 
+@attr.s(slots=True, repr=False)  # pragma: no mutate
+class Request:
+    """Request data extracted from `Case`."""
+
+    method: str = attr.ib()  # pragma: no mutate
+    uri: str = attr.ib()  # pragma: no mutate
+    body: str = attr.ib()  # pragma: no mutate
+    headers: Headers = attr.ib()  # pragma: no mutate
+
+    @classmethod
+    def from_case(cls, case: Case, session: requests.Session) -> "Request":
+        """Create a new `Request` instance from `Case`."""
+        base_url = case.base_url or "http://localhost"
+        kwargs = case.as_requests_kwargs(base_url)
+        request = requests.Request(**kwargs)
+        prepared = session.prepare_request(request)  # type: ignore
+        return cls.from_prepared_request(prepared)
+
+    @classmethod
+    def from_prepared_request(cls, prepared: requests.PreparedRequest) -> "Request":
+        """A prepared request version is already stored in `requests.Response`."""
+        body = getattr(prepared, "raw", getattr(prepared, "body", None))
+
+        if not body:
+            body = ""
+
+        if hasattr(body, "encode"):
```

Comment:
do we always have it? if yes, then we can encode it right away. And also, it will be nice to test the handling of bodies that are not utf-8 encodable